### ISSUE TITLE
another case where secret indexing doesn't seem to work

### ISSUE THREAD

**Issue by linzzzzzz** [secretflow/spu#665]
### Issue Type

Bug

### Source

binary

### Secretflow Version

1.4.0b0

### OS Platform and Distribution

Linux

### Python version

3.8.19

### What happend and What you expected to happen.

```shell
Got below runtime error when running the code in both simulation as well in production. Seems like secret indexing will fail when we do indexing on both dimension for a array with shape [1, x].


RuntimeError: what: 
	[Enforce fail at libspu/core/ndarray_ref.cc:300] (end_indices[idx] <= shape()[idx]). Slice end at axis 0 = 2 is larger than input shape 1
Stacktrace:
#0 spu::kernel::hal::slice()+0x7f0fab0a693e
secretflow/secretflow#1 spu::kernel::hlo::SecretDynamicSlice()+0x7f0faafbc6f2
secretflow/secretflow#2 spu::kernel::hlo::DynamicSlice()+0x7f0faafbd540
secretflow/secretflow#3 spu::device::pphlo::dispatchOp<>()+0x7f0faa9b63fa
secretflow/secretflow#4 spu::device::pphlo::dispatchOp<>()+0x7f0faa9b80e4
secretflow/secretflow#5 spu::device::pphlo::dispatchOp<>()+0x7f0faa9b9f93
secretflow/secretflow#6 spu::device::pphlo::dispatchOp<>()+0x7f0faa9bbfe9
secretflow/secretflow#7 spu::device::pphlo::dispatchOp<>()+0x7f0faa9be2f0
secretflow/secretflow#8 spu::device::runBlock()+0x7f0faaaf9f05
secretflow/secretflow#9 spu::device::runRegion()+0x7f0faaafbf93
secretflow/secretflow#10 spu::device::executeImpl()+0x7f0faa4e27f0
secretflow/secretflow#11 spu::RuntimeWrapper::Run()+0x7f0fa9cc710c
secretflow/secretflow#12 pybind11::cpp_function::initialize<>()::{lambda()#3}::_FUN()+0x7f0fa9c9b5b5
secretflow/secretflow#13 pybind11::cpp_function::dispatcher()+0x7f0fa9c9085e
secretflow/secretflow#14 PyCFunction_Call+0x4f5412
```

### Reproduction code to reproduce the issue.

```shell
sim = spsim.Simulator.simple(2, spu_pb2.ProtocolKind.CHEETAH, spu_pb2.FieldType.FM64)

rand_array = np.random.rand(1, 5)
rand_array_i = [0]

def fn_test(arr1, arr2):

    result = []

    for i in arr2:
        feature, split_point_idx = jnp.divmod(i, 2)
        x = arr1.at[feature.astype(jnp.int8), split_point_idx.astype(jnp.int8)].get()


        result.append(x)

    return result

result = spsim.sim_jax(sim, fn_test)(rand_array, rand_array_i)
```

**Comment by anakinxc** [comment]
Hi @linzzzzzz 

Thanks for reporting, will take a look.

**Comment by tpppppub** [comment]
Hi @linzzzzzz , this issue has been fixed in the recent SPU versions (>=0.8.0b0) but python 3.8 is no longer supported. You can try with the latest SPU.
